fetch_models.js

```diff
@@ -0,0 +1,163 @@
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/text-encoder.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/sd-unet-v1.5-model-b2c4h64w64s77-float16-compute-and-inputs-layernorm.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/Stable-Diffusion-v1.5-vae-decoder-float16-fp32-instancenorm.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/safety-checker.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b_01ec64.encoder-fp16.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b_01ec64.decoder-fp16.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b-encoder-int8.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b-decoder-int8.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm_gelu_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm_gelu_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_encoder_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_encoder_lm_fp16_layernorm_gelu.onnx",
+      path: "./demos/whisper-base/models"
+    }
+  ];
+
+  const downloadFile = async (url, outputPath, retries = 2) => {
+    // // Proxy configuration
+    // const proxyHost = '';
+    // const proxyPort = 33630;
```

Comment:
"33630" - is there any significance to this port? It's not a common one like 80 or 8080 or 21.

Reply:
It was set as a random number, use 8080 instead now.